cranelift/codegen/src/data_value.rs

```diff
@@ -104,6 +105,7 @@ impl DataValue {
             types::I16 => DataValue::I16(i16::from_le_bytes(src[..2].try_into().unwrap())),
             types::I32 => DataValue::I32(i32::from_le_bytes(src[..4].try_into().unwrap())),
             types::I64 => DataValue::I64(i64::from_le_bytes(src[..8].try_into().unwrap())),
+            types::I128 => DataValue::I128(i128::from_ne_bytes(src[..16].try_into().unwrap())),
```

Comment:
Why is this different (`ne` rather than `le`)? It would be useful to have a comment here explaining why we need to use native endian rather than explicit little endian as in the other cases.

Reply:
We actually do need `ne` in the other ones (for big endian systems), however changing `le` -> `ne` is already in a separate PR (#3329).

Reply:
Ah, yes, this makes sense. Thanks for the nudge -- I re-read the issue and decided to merge #3329 to resolve the immediate regression on s390x (so this line now matches what's on `main`), and created a new issue to track the broader question of endianness in the interpreter.